src/wms/src/main/java/org/geoserver/wms/capabilities/GetCapabilitiesTransformer.java

```diff
@@ -1488,121 +1485,87 @@ protected void handleAttribution(PublishedInfo layer) {
-                }
+            attrs.addAttribute("", "height", "height", "", String.valueOf(legendHeight));
 
-                AttributesImpl attrs = new AttributesImpl();
-                attrs.addAttribute("", "width", "width", "", String.valueOf(legendWidth));
-
-                attrs.addAttribute("", "height", "height", "", String.valueOf(legendHeight));
-
-                start("LegendURL", attrs);
-
-                element("Format", defaultFormat);
-                attrs.clear();
-
-                String layerName = layer.prefixedName();
-                Map<String, String> params =
-                        params(
-                                "request",
-                                "GetLegendGraphic",
-                                "format",
-                                defaultFormat,
-                                "width",
-                                String.valueOf(GetLegendGraphicRequest.DEFAULT_WIDTH),
-                                "height",
-                                String.valueOf(GetLegendGraphicRequest.DEFAULT_HEIGHT),
-                                "layer",
-                                layerName);
-                if (style != null) {
-                    params.put("style", style.getName());
-                }
-                String legendURL = buildURL(request.getBaseUrl(), "wms", params, URLType.SERVICE);
+            start("LegendURL", attrs);
 
-                attrs.addAttribute("", "xmlns:xlink", "xmlns:xlink", "", XLINK_NS);
-                attrs.addAttribute(XLINK_NS, "type", "xlink:type", "", "simple");
-                attrs.addAttribute(XLINK_NS, "href", "xlink:href", "", legendURL);
-                element("OnlineResource", null, attrs);
+            element("Format", defaultFormat);
+            attrs.clear();
 
-                end("LegendURL");
+            String layerName = layer.prefixedName();
+            Map<String, String> params =
+                    params(
+                            "request",
+                            "GetLegendGraphic",
+                            "format",
+                            defaultFormat,
+                            "width",
+                            String.valueOf(GetLegendGraphicRequest.DEFAULT_WIDTH),
+                            "height",
+                            String.valueOf(GetLegendGraphicRequest.DEFAULT_HEIGHT),
+                            "layer",
+                            layerName);
+            if (style != null) {
+                params.put("style", style.getName());
             }
+            String legendURL = buildURL(request.getBaseUrl(), "wms", params, URLType.SERVICE);
+
+            attrs.addAttribute("", "xmlns:xlink", "xmlns:xlink", "", XLINK_NS);
+            attrs.addAttribute(XLINK_NS, "type", "xlink:type", "", "simple");
+            attrs.addAttribute(XLINK_NS, "href", "xlink:href", "", legendURL);
+            element("OnlineResource", null, attrs);
```

Comment:
The logic is getting duplicated between transformers, may be worth some code reuse.

Reply:
that functionality still holds and untouched, the unit tests assert that